### PR TITLE
SDK-447 -- Full Windows Sample App

### PR DESCRIPTION
**Pull request by apeterson-branch**
## Reference
SDK-447 -- Full Windows Sample App

## Description
Having "Hello World" will get us only so far.  We want to have a full-blown Windows Sample that shows all of the functionality of the C++ SDK

- Include Deep Linking
- Include Attribution

This is the "Pick A Color" application, as it will then be compatible with Android's HelloBranch.

## Testing Instructions
* Sample app should compile and run
* Using the links in the README, the colors should change as appropriate.

I tested both the Win32 and Win64 versions.

## Risk Assessment [`LOW`]
There is no change to the Core SDK, this is just a sample application.

- [x] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in MR title
- Correctness & Style
    - [ ] Conforms to [Style Guides](https://google.github.io/styleguide/javaguide.html)
    - [ ] Mission critical pieces are documented in code and out of code as needed
- [ ] Unit Tests reviewed and test issue sufficiently
- [ ] Functionality was reviewed in QA independently by another engineer on the team (clone for each required reviewer)

### CODE REVIEW

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+#include "commdlg.h"
+
+#include <BranchIO/Branch.h>
+#include <BranchIO/Event/StandardEvent.h>
+
+#define MAX_LOADSTRING 100
+
+// Global Variables:
+HINSTANCE hInst;                                // current instance
+HWND hwndMain;									// The Main Window
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: formatting of "// The Main Window"

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+            TranslateMessage(&msg);
+            DispatchMessage(&msg);
+        }
+    }
+
+    return (int) msg.wParam;
+}
+
+//
+//  FUNCTION: MyRegisterClass()
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: include the param type: `//  FUNCTION: MyRegisterClass(HINSTANCE)`

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+//  WM_COMMAND  - process the application menu
+//  WM_PAINT    - Paint the main window
+//  WM_DESTROY  - post a quit message and return
+//
+LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    switch (message)
+    {
+	case WM_CREATE:
+		// TStart the Branch Session
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: typo: "TStart"

File: BranchSDK-Samples/Windows/ColorPicker/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+The window background is a nice shade of blue by default.
+
+Further, a Color Dialog will show when the application is first opened allowing you to
+pick a different background color.  Selecting a new color will send an event to Branch
+inidicating "TUTORIAL_COMPLETE".
+
+### Advanced Usage
+
+Clicking on a branch link prior to opening the application can show what happens when the application
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/README.md]: I'd remove "prior to opening the application" for clarity unless I'm missing something (i.e. deeplink doesn't work when the app is in the background)

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp (hunk 1 above)
Comment (apeterson-branch) [reply BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: Tabs vs. Spaces.  Reformatted entire document.

File: BranchSDK-Samples/Windows/ColorPicker/README.md (hunk 4 above)
Comment (bklastaitis-branch) [reply BranchSDK-Samples/Windows/ColorPicker/README.md]: Gotcha!

**Pull request merged**